driver/src/price_finding/naive_solver.rs

```diff
@@ -0,0 +1,74 @@
+use crate::models;
+
+use super::error::PriceFindingError;
+use super::price_finder_interface::Solution;
+use web3::types::U256;
+use crate::models::Order;
+
+impl Order {
```

Comment:
Intuitively, I would have written these impl functions into the models:Order file, so that we do not forget about them and do not write these functions twice.

But I am not so sure here,

Reply:
I was planning to put them in the appropriate place at some point, but at the moment I am just trying to make it work.. Good question though, maybe it actually makes sense to leave them here if they are only ever used here. We will discuss and think about this soon.

Reply:
The calculation of balances is logic that is the same for all solvers. That’s why I’d argue it should live outside the actual solver component (it’s currently part of `run_auction_listener`)

Reply:
Right, yes, although I may have misunderstood @josojo's suggestion here is to pass account balances into this function so that we know what we are about to execute was actually a valid order (i.e. the user has sufficient balance to fulfill the executed sell amount).

Reply:
@fleupold  I am not sure what the reference to "balances" in your comment has to do with these Order implementations. Most of the functions implemented here are specific to the execution of the Naive Solver. However, the Order::surplus function might be useful to have outside of this file.

Reply:
The comment was meant as a  response to the suggestion below where Alex said we should do balance updates inside the solver.

Reply:
@fleupold It's fine if we have the logic for balance updates outside of the solver. However, I was concerned that solvers need to enforce that all orders and their volumes are covered by sufficient balances of their participants. This needs to be in the solver logic itself, doesn't it? 
Also, each solver would hopefully have its own strategy to find a solution that is in accordance with the users' balances.